File: lab2no2.py
def illustration(banki, ostatok, count_steps):
    if count_steps == 0:
        print('Дано')
        print('F' * banki)
    elif count_steps % 2 != 0:
        print(count_steps, ' шаг')
        print('E' * banki)
    elif count_steps % 2 == 0:
        print(count_steps, ' шаг')
        print('F' * (banki - ostatok))
        print('E' * ostatok)

print('Введите количество банок: ')
banki = int(input())
print('Введите курс обмена: ')
obmen = int(input())
if banki <= 0 or obmen <= 0:
    print('Error')
else:
    count_empty = count_steps = ostatok = 0
    count_extra = count_banki = 0
    while banki >= obmen:
        if count_steps == 0: illustration(banki, ostatok, count_steps)
        count_steps += 1
        illustration(banki, ostatok, count_steps)
        count_empty += banki - ostatok
        ostatok = banki % obmen
        banki = (banki // obmen) + (ostatok)
        count_steps += 1
        illustration(banki, ostatok, count_steps)

    if banki < obmen:
        count_empty += banki - ostatok
        count_steps += 1
        illustration(banki, ostatok, count_steps)
        print('Количество выпитых банок: ', count_empty)
        print('Количество шагов: ', count_steps)

File: lab2no2.js
function illustration(banki, ostatok, countSteps) {
    if (countSteps === 0) {
        console.log("Jars");
        console.log("F".repeat(banki));
    } else if (countSteps % 2 !== 0) {
        console.log("E".repeat(banki));
    } else {
        console.log("F".repeat(banki - ostatok));
        console.log("E".repeat(ostatok));
    }
}

function main() {
    const readline = require('readline');
    const rl = readline.createInterface({
        input: process.stdin,
        output: process.stdout
    });

    rl.question("Input jar quantity: ", function(banki) {
        rl.question("Input exchange: ", function(obmen) {
            banki = parseInt(banki);
            obmen = parseInt(obmen);

            if (banki <= 0 || obmen <= 0) {
                console.log("Error");
            } else {
                let countEmpty = 0;
                let countSteps = 0;
                let ostatok = 0;
                let newBanki = banki;

                while (newBanki >= obmen) {
                    if (countSteps === 0) {
                        illustration(newBanki, ostatok, countSteps);
                    }
                    countSteps++;
                    illustration(newBanki, ostatok, countSteps);
                    countEmpty += newBanki - ostatok;
                    ostatok = newBanki % obmen;
                    newBanki = Math.floor(newBanki / obmen) + ostatok;
                    countSteps++;
                    illustration(newBanki, ostatok, countSteps);
                }

                if (newBanki < obmen) {
                    countEmpty += newBanki - ostatok;
                    countSteps++;
                    illustration(newBanki, ostatok, countSteps);
                    console.log(countEmpty);
                    console.log(countSteps);
                }
            }

            rl.close();
        });
    });
}

main();
